api/app/models/user.rb

```diff
@@ -4,4 +4,14 @@ class User < ApplicationRecord
 
   # Validations
   validates :name, :birth_date, presence: true
+  validate :password_complexity
+
+  private
+
+  def password_complexity
+    return if password =~ /(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-])/
+
+    errors.add :password, 'Complexity requirement not met. ' \
+                          'Please use: 1 uppercase, 1 lowercase, 1 digit and 1 special character'
```

Comment:
Esto se devuelve al usuario? Se que el codigo esta en ingles, pero los mensajes deberian estar en español

Reply:
En realidad si esto se va a controlar desde el FE, no necesariamente va a ser visible para el end-user. Van a ver lo que los gurises agreguen en el FE.

De todas formas sí, quizás estaría bueno dejarlo en español por las dudas. Lo había pensado, pero no le puse mucho pienso en el wording. Alguna idea? Algo como esto quizás?:

```
Contraseña inválida. Debe incluir al menos una mayúscula, una minúscula, 1 dígito y un carácter especial.
```

Reply:
En el front teniamos pensado crear un archivo con un mappeo entre los mensajes en ingles y en español para mostrarlos. La otra opcion es enviarlo directo del back al front en español.
Las dos opciones se pueden implementar bien es definir nomas cual queremos elegir, si quieren lo dejamos en ingles para tener consistencia con el codigo.

Reply:
Perdon pero para aclarar. La idea es simplemente mantener un archivo con los strings de la aplicacion para mantener consistencia con mensajes de error etc. Por ahora en espanol.

Reply:
En el caso de errores del back, el front necesita saber que tipo de error es para saber que mostrar al usuario, no es lo mismo un error por password que uno por mail ya creado, o algo por el estilo. Con eso alcanza.

Algun tipo de convencion estaria bueno usar pero lo podemos ver mas adelante. Pero creo que por ahora estaria bueno en enfocarse en tener algo funcional. Luego podemos crear tickets para mejorar ese tipo de cosas.

Reply:
Como quieran, la idea era que no le saliera un server not found al usuario, pero si es muy complejo devolvemos el error del back nomas dale.

Reply:
Por ejemplo en mobile un mensaje de error muy largo capaz te complica el estilo.

Reply:
Dale genial, se puede ver despues igual.